pkg/provisioner/ironic/ironic.go

```diff
@@ -1001,19 +1049,50 @@ func (p *ironicProvisioner) buildManualCleaningSteps(bmcAccess bmc.AccessDetails
 	}
 	cleanSteps = append(cleanSteps, raidCleanSteps...)
 
-	// Build bios clean steps
-	settings, err := bmcAccess.BuildBIOSSettings(data.FirmwareConfig)
+	// Get the subset (currently 3) of vendor specific BIOS settings converted from common names
+	bmcsettings, err := bmcAccess.BuildBIOSSettings(data.FirmwareConfig)
 	if err != nil {
 		return nil, err
 	}
-	if len(settings) != 0 {
+
+	var newSettings []map[string]string
+	if data.ActualFirmwareSettings != nil {
+		// If we have the current settings from Ironic, update the settings to contain:
+		// 1. settings converted by BMC drivers that are different than current settings
+		for _, bmcsetting := range bmcsettings {
+			if val, exists := data.ActualFirmwareSettings[bmcsetting["name"]]; exists {
+				if bmcsetting["value"] != val {
+					newSettings = buildFirmwareSettings(newSettings, bmcsetting["name"], bmcsetting["value"])
+				}
+			} else {
+				p.log.Info("name converted from bmc driver not found in firmware settings", "name", bmcsetting["name"])
```

Comment:
Please log BMH and/or Node UUID

Reply:
For includeSchema - yes its always true now as we always get the settings and full schema. i'd like to keep this logic though as we may find in the future there are situations where we just want to get the settings

Reply:
> Please log BMH and/or Node UUID
Added nodeID